sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java

```diff
@@ -102,7 +103,12 @@ public String getHost() {
      */
     public BlobUrlParts setHost(String host) {
         this.host = host;
-        this.isIpUrl = ModelHelper.IP_V4_URL_PATTERN.matcher(host).find();
+        try {
+            this.isIpUrl = ModelHelper.determineAuthorityIsIpStyle(host);
+        } catch (MalformedURLException e) {
+            throw logger.logExceptionAsError(new IllegalStateException("Authority is malformed. Host: "
+                + host));
+        }
```

Comment:
Does this issue affect Queues as well? I know it also supports IP-style URLs in Azurite.

Reply:
Yea I should probably bring it over there as well. Thanks for thinking of that.